core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +82,44 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
                 .collect(Collectors.toList());
     }
 
+    /** 
+     *  @return Data for market depth chart  
+    */
+    public MarketDepthInfo getMarketDepth(String currencyCode) {        
+        List<OpenOffer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode); 
+        List<OpenOffer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode); 
+
+        // Create buyer hashmap {key:price, value:count}
+        HashMap<Double,Long> buyHM = new HashMap<Double,Long>();        
+        buyOffers.forEach((openOffer) -> {
+            Double price = openOffer.getOffer().getOfferPayload().getCoinAmount();
+            Long previousValue = buyHM.get(price) != null ? buyHM.get(price) : Long.valueOf(0);
+            buyHM.put(price, previousValue + 1);
+        });
+        
+        // Create seller hashmap {key:price, value:count}
+        HashMap<Double,Long> sellHM = new HashMap<Double,Long>();
+        sellOffers.forEach((openOffer) -> {
+            Double price = openOffer.getOffer().getOfferPayload().getCoinAmount();
+            Long previousValue = sellHM.get(price) != null ? sellHM.get(price) : Long.valueOf(0);
+            sellHM.put(price, previousValue + 1);
+        });
+        
+        // Use TreeMap to sort by key (asc)
+        TreeMap<Double,Long> buyTM = new TreeMap<Double,Long> (buyHM);  
+        TreeMap<Double,Long> sellTM = new TreeMap<Double,Long> (sellHM);  
```

Comment:
you could use TreeMap directly in the original map instead of HashMap

Reply:
will do thanks

Reply:
done